src/service/resources-impl.js

```diff
@@ -336,20 +336,30 @@ export class Resources {
       element.id = 'AMP_' + resource.getId();
     }
     this.resources_.push(resource);
+    this.buildOrScheduleBuildForElement_(element);
+    dev().fine(TAG_, 'element added:', resource.debugid);
+  }
 
+  /**
+   * Builds the element if ready to be built, otherwise adds it to pending resources.
+   * @private
+   */
+  buildOrScheduleBuildForElement_(element) {
+    const resource = this.getResourceForElement(element);
     if (this.isRuntimeOn_) {
+      if (element.isBuilt()) {
```

Comment:
You meant on `upgraded` not `add`, right? 

I was thinking the `schedulePass` inside this new function would take care of that, but I guess it's not the same, putting it back in upgraded call.

Is there harm from always scheduling a pass at the end of `buildOrScheduleBuildForElement_` but inside the `if(isRuntimeOn)` check instead?